safe/gui/tools/batch/batch_dialog.py

```diff
@@ -271,99 +280,252 @@ def run_script(self, filename):
+        # get extension and basename to create layer
+        base_name, extension = os.path.splitext(file_name)
 
-            extent = QgsRectangle(*coordinates)
+        # check if raster or vector layer from extension
+        if extension in ['.asc', '.tif']:
+            layer = QgsRasterLayer(full_path, base_name)
+            if layer.isValid():
+                return layer
+            else:
+                LOGGER.critical('Failed to create layer from scenario')
+                return
+        elif extension in ['.shp']:
+            layer = QgsVectorLayer(full_path, base_name, 'ogr')
+            if layer.isValid():
+                return layer
+            else:
+                LOGGER.critical('Failed to create layer from scenario')
+                return
+        else:
+            LOGGER.warning('Input extension in scenario is not recognized')
+            return
 
-            self.dock.define_user_analysis_extent(extent, crs)
+    def run_task(self, task_item, status_item, count=0, index=''):
+        """Run a single task.
 
-            message = 'set layer extent to %s ' % extent.asWktCoordinates()
-            # LOGGER.info(message)
+        :param task_item: Table task_item containing task name / details.
+        :type task_item: QTableWidgetItem
 
-            self.iface.mapCanvas().setExtent(extent)
+        :param status_item: Table task_item that holds the task status.
+        :type status_item: QTableWidgetItem
 
-        status, message = scenario_runner.run_scenario(self.dock)
+        :param count: Count of scenarios that have been run already.
+        :type count:
 
-        return status, message
+        :param index: The index for the table item that will be run.
+        :type index: int
 
-    def reset_status(self):
-        """Set all scenarios' status to empty in the table
+        :returns: Flag indicating if the task succeeded or not.
+        :rtype: bool
         """
-        for row in range(self.table.rowCount()):
-            status_item = self.table.item(row, 1)
-            status_item.setText(self.tr(''))
+
+        self.enable_busy_cursor()
+        for layer_group in self.layer_group_container:
+            layer_group.setVisible(False)
+
+        # set status to 'running'
+        status_item.setText(self.tr('Running'))
+
+        # .. see also:: :func:`appendRow` to understand the next 2 lines
+        variant = task_item.data(QtCore.Qt.UserRole)
+        value = variant[0]
+        result = True
+
+        if isinstance(value, str):
+            filename = value
+            # run script
+            try:
+                self.run_script(filename)
+                # set status to 'OK'
+                status_item.setText(self.tr('Script OK'))
+            except Exception as e:  # pylint: disable=W0703
+                # set status to 'fail'
+                status_item.setText(self.tr('Script Fail'))
+
+                LOGGER.exception('Running macro failed. The exception: ' +
```

Comment:
We try having as much as possible on the left. It's more readable if  the sentence is on one line only.